omnigibson/object_states/particle_modifier.py

```diff
@@ -654,7 +821,9 @@ def _initialize(self):
         system = get_system(system_name)
 
         if self.method == ParticleModifyMethod.PROJECTION and self._projection_mesh_params["visualize"]:
-            radius, height = self._projection_mesh_params["extents"][0] / 2.0, self._projection_mesh_params["extents"][2]
+            assert self._projection_mesh_params["type"] in {"Cylinder", "Cone"}, \
+                f"{self.__class__.__name__} visualization only supports Cylinder and Cone types!"
+            radius, height = np.mean(self._projection_mesh_params["extents"][:2]) / 2.0, self._projection_mesh_params["extents"][2]
```

Comment:
is np.mean necessary? For "Cylinder" and "Cone", isn't `self._projection_mesh_params["extents"][0] == self._projection_mesh_params["extents"][1]`?

Reply:
Two remaining issues:
- Check if meta meshes exist (link_prefix_0_0/mesh_0)
- Make sure meta meshes for ToggledOn and ParticleApplier are not stretched into different proportions when the object is scaled by different amount. Let's revisit this to see if this is still a good idea or not.

Reply:
- Support cube and sphere when creating visual meshes
- Set scale for these meshes based on extents, rather than setting size/radius/height

Reply:
Add an additional comment:

```
    # Scale radius and height by the parent scale -- projection always points in the negative-z direction of the
    # parent frame
    # We do this AFTER we create the source sphere because the actual projection is scaled in the world frame, whereas
    # the source sphere is already scaled by its own parent frame
    projection_radius *= np.mean(parent_scale[:2])
    projection_height *= parent_scale[2]
```

saying the visualization will NOT match the meta mesh if the parent scale non-uniformally

Reply:
done